packages/bot/src/features/move.ts

```diff
@@ -0,0 +1,171 @@
+ */
+export class MoveFeature extends BaseFeature {
+	constructor(context: FeatureContext) {
+		super(context);
+	}
+
+	/**
+	 * Handles the start of move operation from a select menu interaction
+	 * Note: Unlike Avatar and Rename, Move doesn't require the db parameter
+	 * as it only checks moderator permissions via guild member cache
+	 */
+	async start(): Promise<void> {
+		const interaction = this.interaction as Djs.StringSelectMenuInteraction;
+		const moderator = interaction.guild?.members.cache
+			.get(this.interactionUser.id)
+			?.permissions.has(Djs.PermissionsBitField.Flags.ManageRoles);
+		if (moderator) await this.showMove(interaction);
+		else
+			await interaction.reply({
+				content: this.ul("modals.noPermission"),
+				flags: Djs.MessageFlags.Ephemeral,
+			});
+	}
+
+	/**
+	 * Displays a modal for selecting a user to move the character to
+	 */
+	private async showMove(interaction: Djs.StringSelectMenuInteraction): Promise<void> {
+		const modal = new Djs.ModalBuilder()
+			.setCustomId("move")
+			.setTitle(this.ul("button.user"))
+			.addLabelComponents((label) =>
+				label
+					.setLabel(this.ul("common.user"))
+					.setUserSelectMenuComponent((select) =>
+						select.setCustomId("user").setRequired(true).setMaxValues(1)
+					)
+			);
+		await interaction.showModal(modal);
+	}
+
+	/**
+	 * Handles a Discord modal submission to validate and process the transfer of a character between users within a guild.
+	 *
+	 * Validates user input, retrieves and updates character ownership, and invokes the move command to complete the transfer. Provides localized error feedback and resets the interaction state if validation fails at any step.
+	 */
+	async validate(): Promise<void> {
+		const interaction = this.interaction as Djs.ModalSubmitInteraction;
+		if (!interaction.message || !interaction.channel || !interaction.guild) return;
+		if (!this.client) return;
+		profiler.startProfiler();
+		const message = await (interaction.channel as TextChannel).messages.fetch(
+			interaction.message.id
+		);
+		await interaction.deferReply({ flags: Djs.MessageFlags.Ephemeral });
+		const user = interaction.fields.getSelectedUsers("user")?.first();
+		if (!user) return;
+		const embed = getEmbeds(message, "user");
+		if (!embed) throw new BotError(this.ul("error.embed.notFound"), botErrorOptions);
+
+		const oldUserId = embed
+			.toJSON()
+			.fields?.find((field) => findln(field.name) === "common.user")
+			?.value.replace(/<@|>/g, "");
+		if (!oldUserId) {
+			await interaction.reply({
+				embeds: [embedError(this.ul("error.user.notFound"), this.ul)],
+				flags: Djs.MessageFlags.Ephemeral,
+			});
+			return await resetButton(message, this.ul);
+		}
+		const oldUser = await isUserNameOrId(oldUserId, interaction);
+		if (!oldUser) {
+			await interaction.reply({
+				embeds: [embedError(this.ul("error.user.notFound"), this.ul)],
+				flags: Djs.MessageFlags.Ephemeral,
+			});
+			return await resetButton(message, this.ul);
```

Comment:
TS2322: Type OmitPartialGroupDMChannel<Message<boolean>> is not assignable to type void